src/snowflake/snowpark/modin/plugin/io/snow_io.py

```diff
@@ -593,28 +593,18 @@ def read_gbq(
         pass  # pragma: no cover
 
     @classmethod
-    @pandas_module_level_function_not_implemented()
-    def read_html(
-        cls,
-        io,
-        *,
-        match=".+",
-        flavor=None,
-        header=None,
-        index_col=None,
-        skiprows=None,
-        attrs=None,
-        parse_dates=False,
-        thousands=",",
-        encoding=None,
-        decimal=".",
-        converters=None,
-        na_values=None,
-        keep_default_na=True,
-        displayed_only=True,
-        **kwargs,
-    ):
-        pass  # pragma: no cover
+    def read_html(cls, **kwargs) -> list[SnowflakeQueryCompiler]:
+        """
+        Read HTML tables into a list of query compilers.
+        """
+        return [cls.from_pandas(df) for df in pandas.read_html(**kwargs)]
+
+    @classmethod
+    def read_xml(cls, **kwargs) -> SnowflakeQueryCompiler:
```

Comment:
To prevent xml entity resolution security problems I would throw an error if:
* storage_options is used as a parameter
* if parser is changed from the default, lxml
* if stylesheet is used
* and /possibly/ if namespaces is used.

I know pandas supports all these but we can add support in later if need be and we don't have to deal with that class of issue

Reply:
John and I discussed offline. Below is the summary of our convo.

1- Even if we blocked some of the native pandas functionality when directly calling Snowpark pandas, users can still access that same blocked functionality by simply calling native pandas themselves and then passing the resulting df into Snowpark pandas:

```
native_df = pandas.read_*(<some_param_combination>)
df = pd.DataFrame(native_df)
```

2- We are exposing the exact same signatures as native pandas, and furthermore we are explicitly mentioning in our docs that we are using native pandas internally to support those methods. So the user should not be surprised by any behavior because of using specific native pandas functionalities that they explicitly invoked.

3- Based on the above, we believe the liability should not be on us (Snowflake) but rather on the end users in those cases (similar to the case when users are simply using native pandas without using Snowpark pandas at all).

@sfc-gh-dolee would love to get your input as well.

Reply:
@sfc-gh-jkew I checked with Doris and she agrees with the above.

In her words:
> I agree with your points in the comments. I don’t see how this would pose any additional security risk over what we already support with native pandas today.